manifest/docker_schema2_list.go

```diff
@@ -86,6 +86,30 @@ func (list *Schema2List) UpdateInstances(updates []ListUpdate) error {
 	return nil
 }
 
+func (list *Schema2List) UpdateInstancesAndDiscardOthers(listUpdate []ListUpdate) error {
+	existingManifest := map[digest.Digest]Schema2ManifestDescriptor{}
+	for _, m := range list.Manifests {
+		for _, u := range listUpdate {
+			if m.Digest == u.Digest {
+				existingManifest[u.Digest] = m
+			}
+		}
+	}
+	list.Manifests = []Schema2ManifestDescriptor{}
+	for _, u := range listUpdate {
+		updatedManifest, ok := existingManifest[u.Digest]
```

Comment:
I don’t think this works at all in the case where the copy had to convert the format of the original manifest, so `u.Digest` doesn‘t match any pre-existing value in `list.Manifests`.